pkg/frontend/admin_openshiftcluster_vmresize.go

```diff
@@ -45,44 +42,30 @@ func (f *frontend) _postAdminOpenShiftClusterVMResize(log *logrus.Entry, ctx con
 		return err
 	}
 
-	k, err := f.kubeActionsFactory(log, f.env, doc.OpenShiftCluster)
+	// checks if the Virtual machines exists in the Cluster RG
+	exists, err := action.ResourceGroupHasVM(ctx, vmName)
 	if err != nil {
 		return err
 	}
-
-	nodeList, err := k.KubeList(ctx, "node", "")
-	if err != nil {
-		return err
-	}
-
-	var u unstructured.Unstructured
-	var nodes corev1.NodeList
-	if err = json.Unmarshal(nodeList, &u); err != nil {
-		return err
-	}
-
-	err = kruntime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &nodes)
-	if err != nil {
-		return err
-	}
-
-	nodeExists := false
-	for _, node := range nodes.Items {
-		if _, ok := node.ObjectMeta.Labels["node-role.kubernetes.io/master"]; !ok {
-			continue
-		}
-
-		if strings.EqualFold(vmName, node.ObjectMeta.Name) {
-			nodeExists = true
-			break
-		}
-	}
-
-	if !nodeExists {
+	if !exists {
 		return api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeNotFound, "",
-			`"The master node '%s' under resource group '%s' was not found."`,
+			`"The VirtualMachine '%s' under resource group '%s' was not found."`,
 			vmName, resourceGroupName)
 	}
 
+	if !nodeIsMaster(vmName) {
+		return api.NewCloudError(http.StatusForbidden, api.CloudErrorCodeForbidden, "",
+			`"The vmName '%s' provided cannot be resized. It is either not a master node or not adhering to the standard naming convention."`,
+			vmName)
+	}
+
 	return action.VMResize(ctx, vmName, vmSize)
 }
+
+// A bland check, to validate if the node is master by checking vmName ends with pattern "-master-[0-9]"
+// return false, when the node is either not a master node or not adhering to the standard naming convention.
+// return true, if regexp satisfies
+func nodeIsMaster(vmName string) bool {
+	r := regexp.MustCompile(`.*-master-[0-9]{1,}$`)
```

Comment:
this isn't really required tbh. It's used only once right there

Reply:
Yes. I agree it is not really required (that's why it was a LGTM + comment) but I think it helps in the long term because if the regex changes in the future, the function (_nodeIsMaster_) will not be touched, just the const on top of the file. 😃